src/grpc/stream_infer_handler.cc

```diff
@@ -515,14 +521,17 @@ ModelStreamInferHandler::Finish(InferHandler::State* state)
   // entire stream. Otherwise just finish this state.
   if (state->context_->IsRequestsCompleted()) {
     state->context_->step_ = Steps::COMPLETE;
-    state->step_ = Steps::COMPLETE;
+    state->step_ = Steps::PARTIAL_COMPLETION;
+    LOG_VERBOSE(2) << "Finishing responder from state " << state->unique_id_;
```

Comment:
From PR description:
> There is a possibility that server when shutting down might not receive the Done() event. Exploring more on this,

@tanmayv25 is this resolved? Or still exploring? Can you update PR description to reflect latest coverage/gaps?

Reply:
This is a similar server shutdown problem. The race condition is very specific. However, the good thing is we call `TRITONSERVER_ServerStop()` before shutting down the completion queue. So, only the handler thread will be interacting with the completion queue at this point. Opened DLIS-5563 to address this.

Reply:
Updated the PR description as well.

Reply:
Follow-up is fine, thanks